CMake error in the logic in include/add_pfunit_sources.cmake or callers

Hi @tclune et al.,

There seems to be a logic error somewhere in either `include/add_pfunit_sources.cmake` or one of the functions calling this:

```
CMake Error at include/add_pfunit_sources.cmake:33 (file):
  file problem creating directory:
  /tmp.F90unit-20190507-36706-1qjb902/pFUnit-4.0.0/build/tests/funit-core
Call Stack (most recent call first):
  include/add_pfunit_ctest.cmake:47 (add_pfunit_sources)
  tests/funit-core/CMakeLists.txt:98 (add_pfunit_ctest)
```

Your code is taking the current working directory and substituting `.F90` in the directory path. e.g., 

`/tmp/pfunit-20190507-36706-1qjb902/pFUnit-4.0.0/build` ➡️ `/tmp/pfunit-20190507-36706-1qjb902/pFUnit-4.0.0/build`

I think I know the fix for this and can open a PR. This is a blocker for adding a homebrew formula, so the sooner a minor release can be minted after fixing, the better.

--- a/include/add_pfunit_sources.cmake
+++ b/include/add_pfunit_sources.cmake
@@ -23,8 +23,14 @@ function( ADD_PFUNIT_SOURCES out_var )
          file( RELATIVE_PATH rel_file "${CMAKE_CURRENT_SOURCE_DIR}" "${abs_file}" )
 
          # replace the extension with .F90 to determine the output file name
-	 get_filename_component (extension ${file} EXT)
-         string( REGEX REPLACE "${extension}" ".F90" out_file "${CMAKE_CURRENT_BINARY_DIR}/${rel_file}" )
+	 get_filename_component (file_name "${rel_file}" NAME_WE)
+	 get_filename_component (file_dir "${rel_file}" DIRECTORY)
+	 if(NOT "${file_dir}" STREQUAL "")
+	   set(file_dir "${CMAKE_CURRENT_BINARY_DIR}/${file_dir}")
+	 else()
+	   set(file_dir "${CMAKE_CURRENT_BINARY_DIR}")
+	 endif()
+	 set(out_file "${file_dir}/${file_name}.F90")
          # append the output file to the list of outputs
          list( APPEND out_files "${out_file}" )
          # create the output directory if it doesn't exist

